[Docs] Clarify managed spot jobs against sky launch (#3561)
skypilot-org/skypilot: docs/source/examples/managed-jobs.rst

Managed Spot Jobs
-----------------

> *SkyPilot managed spot job* (:code:`sky jobs launch --use-spot`) automatically finds available spot resources across regions and clouds to maximize availability.

Reviewer: ```suggestion
In this mode, :code:`sky jobs launch --use-spot` is used to launch a managed spot job. SkyPilot automatically finds available spot resources across regions and clouds to maximize availability.
```

Any spot preemptions are automatically handled by SkyPilot without user intervention.

> 

Reviewer: ```suggestion

Quick comparison between _unmanaged spot clusters_ vs. _managed spot jobs_:
```

.. list-table::
   :widths: 30 18 12 35
   :header-rows: 1

   * - Command
     - Managed?
     - SSH-able?
     - Best for
   * - :code:`sky launch --use-spot`
     - Unmanaged spot cluster
     - Yes

>      - Interactive dev on spot instances (e.g., :ref:`SSH, VSCode, Jupyter <dev-connect>`).

Reviewer: ```suggestion
     - Interactive dev on spot instances (best for hardware with low preemption rates)
```

Reviewer: Removing the parenthesis there since we know a bunch of people queue jobs on it.

   * - :code:`sky jobs launch --use-spot`
     - Managed spot job (auto-recovery)
     - No

>      - Scaling out long-running jobs (e.g., data processing, training, batch inference).

Reviewer: ```suggestion
     - Scaling out long-running jobs (e.g., data processing, training, batch inference)
```

SkyPilot automatically finds available spot resources across regions and clouds to maximize availability.
Any spot preemptions are automatically handled by SkyPilot without user intervention.

> .. tip::

Author: Good idea! Changed it accordingly:
<img width="872" alt="image" src="https://github.com/skypilot-org/skypilot/assets/6753189/bcdaa50b-8f72-44ce-a2d9-db3271e29d26">
